SETUP/tests/jsTests/ajaxTests.js

```diff
@@ -0,0 +1,76 @@
+/* global QUnit ajax */
+/* exported codeUrl */
+
+let codeUrl = "https://www.dummy.org";
+
+QUnit.module("Ajax test", function() {
+
+    QUnit.test("Return correct data", function (assert) {
+        function fetchPromise() {
+            const blob = new Blob([JSON.stringify({key: "value"})], {type : 'application/json'});
+            const init = {status: 200, headers: {'content-type': 'application/json'}};
+            return Promise.resolve(new Response(blob, init));
+        }
+
+        return ajax("GET", "myUrl", {}, {}, fetchPromise)
+            .then(function(data) {
+                assert.deepEqual(data, {key: "value"});
```

Comment:
It probably works, but probably a good idea to use the qunit async so the test runner waits for all promises to resolve before marking the test passed or not: https://api.qunitjs.com/assert/async/

Reply:
I don't claim to have much understanding of Qunit. I was following the example at the bottom of this page: https://api.qunitjs.com/QUnit/test/#example-test-with-promise